packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -2834,3 +2882,277 @@ typedef SelectableRegionContextMenuBuilder = Widget Function(
   BuildContext context,
   SelectableRegionState selectableRegionState,
 );
+
+/// Signature for the callback that reports when the user changes the selection
+/// under a [SelectionListener].
+typedef SelectionListenerSelectionChangedCallback = void Function(List<SelectedContentRange<Object>>? selections);
+
+/// A widget that allows the user to listen to selection changes
+/// for the child subtree it wraps under a [SelectionArea] or [SelectableRegion].
+///
+/// This widget should have an ancestor [SelectionArea] or [SelectableRegion]
+/// to be able to listen to selection changes in its subtree.
```

Comment:
By "its" do you mean the SelectionArea/SelectableRegion or the SelectionListener?

Reply:
Also another question:

```dart
SelectionArea(
  child: SelectionListener(
    onSelectionChanged: ...
    child: SelectionArea(
      ...
    ),
  ),
),
```

Will the SelectionListener pick up changes in the nested SelectionArea?

Reply:
It will not because nested SelectionAreas do not cooperate with each other, i.e. a nested SelectionArea will not register it children selectables to an ancestor SelectionRegistrar. I'll add some documentation about this.